aws-android-sdk-cognitoauth/src/androidTest/java/com/amazonaws/mobileconnectors/cognitoauth/LocalDataManagerIntegrationTest.java

```diff
@@ -403,30 +405,17 @@ private void verifySharedPreferencesForCachedSession() {
         assertNotNull(sharedPreferencesForAuth.getString(lastAuthUserKey, null));
     }
 
-    private void verifySharedPreferencesForCachedState(AWSKeyValueStore awsKeyValueStore, String proofKeyValue, String scopesValue) {
+    private void verifySharedPreferencesForCachedState(AWSKeyValueStore awsKeyValueStore, String proofKeyValue, Set scopesValue) {
         assert sharedPreferencesForAuth.getAll().keySet().size() > 0;
         assert sharedPreferencesForAuthEncryptionMaterials.getAll().keySet().size() > 0;
 
-        assertEquals(proofKeyValue,
-                LocalDataManager.getCachedProofKey(awsKeyValueStore, InstrumentationRegistry.getTargetContext(), "key"));
-        assertEquals(scopesValue == null ? new HashSet<String>() : setFromString(scopesValue),
-                LocalDataManager.getCachedScopes(awsKeyValueStore, InstrumentationRegistry.getTargetContext(), "key"));
-    }
+        String cachedProofKey = LocalDataManager.getCachedProofKey(awsKeyValueStore, InstrumentationRegistry.getTargetContext(), "key");
+        Set cachedScopes = LocalDataManager.getCachedScopes(awsKeyValueStore, InstrumentationRegistry.getTargetContext(), "key");
 
-    static String setToString(Set<String> stringSet) {
-        StringBuilder strBuilder = new StringBuilder();
-        int index = 0;
-        for (String str: stringSet) {
-            strBuilder.append(str);
-            if (index++ < stringSet.size() - 1) {
-                strBuilder.append(",");
-            }
-        }
-        return strBuilder.toString();
-    }
+        assertEquals(proofKeyValue, cachedProofKey);
 
-    static Set<String> setFromString(String str) {
-        String[] stringArray = str.split(",");
-        return new HashSet<String>(Arrays.asList(stringArray));
```

Comment:
Is the `if` a good thing? Don’t we want the test to fail if this is unexpectedly `null`?

Reply:
Hmm, not sure. Before, this assertion was testing an empty set against what was in the cache which did not seem sensible to me. Perhaps what would be better is to assert the set is empty if a null is passed for scopes.

Reply:
Let me know if I'm any closer with this revision. In the null case, I'm ensuring cachedScopes if empty, which was the original intent.